File: 프로그래머스/알고리즘 고득점 Kit/완전탐색/전력망을 둘로 나누기.js
function solution(N, wires) {
	if (N === 2) return 0;

	ans = N;
	for (const wire of wires) {
		const sub = wires.filter((e) => e !== wire);
		const array = [sub[0]];
		for (const _ of sub) {
			sub.forEach(([u, v]) => {
				if (array.includes(u)) array.push(v);
				if (array.includes(v)) array.push(u);
			});
		}
		ans = min(ans, Math.abs(N - array.length * 2));
	}
	return ans;
}

File: 프로그래머스/알고리즘 고득점 Kit/완전탐색/전력망을 둘로 나누기.py
from itertools import product

def solution(N, wires):
    if N == 2:
        return 0
    
    ans = N
    for sub in [wires[:i] + wires[i+1:] for i in range(len(wires))]:
        s = set(sub[0])
        [s.update(edge) for _, edge in product(sub, sub) if set(edge) & s]
        ans = min(ans, abs(N-len(s)*2))
    return ans